src/isl/isl_type_reference.rs

```diff
@@ -102,6 +102,11 @@ impl IslTypeRef {
                         "a base or alias type reference can not be null.struct",
                     )
                 }
+                if value.annotations().any(|a| a.text() == Some("nullable")) {
+                    return invalid_schema_error(
+                        "nullable type reference is only allowed for built in types",
+                    )
+                }
```

Comment:
Oh no. I just added this here because I realized the bug while working on `document` type. 
I will try to keep these type of changes separate for future PRs.